pmd-cataloger/src/main/java/sfdc/sfdx/scanner/messaging/SfdxMessager.java

```diff
@@ -0,0 +1,58 @@
+package sfdc.sfdx.scanner.messaging;
+
+import com.google.gson.Gson;
+
+enum MessageType {
+  WARNING
+}
+
+enum MessageHandler {
+  UX
+}
+
+public class SfdxMessager {
+  // The START string gives us something to scan for when we're processing output.
+  private static final String START = "SFDX-START";
+  // The END string lets us know when a message stops, which should prevent bugs involving multi-line output.
+  private static final String END = "SFDX-END";
+
+  private static SfdxMessager INSTANCE = null;
+
+  public static SfdxMessager getInstance() {
+    if (INSTANCE == null) {
+      INSTANCE = new SfdxMessager();
+    }
+    return INSTANCE;
+  }
+
+  public void uxWarn(String msg) {
+    uxWarn(msg, false);
+  }
+
+  public void uxWarn(String msg, boolean verbose) {
+    System.err.println(formatMessage(msg, MessageType.WARNING, MessageHandler.UX, verbose));
```

Comment:
Thinking out loud. If we are about to deal with different tags, would it be more stable to use a known structure like JSON to print all the messages at once? Any other random things appearing in stdout wouldn't have unexpected effects on how we handle the string.
We can add messages to a List or some other structure maintained by this class. And in main(), we can put a top level try-catch-finally and print the messages (as a JSON or at least as a contiguous block) to stdout in the finally block.

Reply:
I think the stability of any option is going to be roughly the same. But you do raise an important question of making sure that the ordering of logs is preserved. I'll make some quick changes to make sure that happens.